src/apps/messages/domain/messagesController.js

```diff
@@ -50,6 +52,28 @@ export const getMessageById = async (req, res, next) => {
   });
 };
 
+export const getMessageByCategory = async (req, res, next) => {
+  const { id } = req.params;
```

Comment:
I noticed your code uses req.params to get the id for filtering messages. It's important to distinguish between params and query in Express. Use req.params for essential data like a specific resource ID (e.g., /messages/:id for a particular message). On the other hand, req.query is better for optional, non-essential data like filters or search criteria (e.g., /messages?category=someCategory). In your case, if id isn't an actual ID but a filter category, consider using req.query instead. This approach keeps your API intuitive and RESTful, making it easier for others to understand and use your endpoints effectively.

so in ur case use query and not params  and use diff naming other than id because that value u r getting doesnt represent id u can use messageType instead or something more appropriate  .
 client side should be changed accordingly when fetching this route/path

Reply:
i get the categoryID and get all messages based on this CategroyID, do you think i should change the params to query? because that is the only thing i expect to receive in the /messages/categroy/ route.

Reply:
if the id u excpect to get is mongo id keep it as param , but isee u checking if the value of the id is 'all' wich is string

Reply:
I am supposed to get all messages with the category id,
There is one edge case where I have a category called "main" where i get the newest 5 messages from all categories, and thats where the check comes in